src/django_oapif/decorators.py

```diff
@@ -113,13 +122,13 @@ def finalize_response(self, request, response, *args, **kwargs):
             def get_queryset(self):
                 qs = super().get_queryset()
 
-                if USE_PG_GEOJSON:
-                    # NOTE the defer should not be needed, as the field should be skipped already when we define `Serializer.Meta.Fields` without the `geom` col
-                    qs = qs.defer("geom")
-                    qs = qs.annotate(geojson=Cast(AsGeoJSON("geom", False, False), models.JSONField()))
+                if geom_db_serializer and geom_field:
```

Comment:
why the two conditions? Also the naming is a bit strange with `_geom_json_db`, why don't we do `_geom_geosjon` only?

Reply:
well, serialization on db is True by default and you don't want to go there if there is no geometry field